pyserini/search/_searcher.py

```diff
@@ -248,4 +250,55 @@ def bm25(k1=0.9, b=0.4):
 
     @staticmethod
     def qld(mu=1000):
-        return autoclass('org.apache.lucene.search.similarities.LMDirichletSimilarity')(mu)
\ No newline at end of file
+        return autoclass('org.apache.lucene.search.similarities.LMDirichletSimilarity')(mu)
+
+
+class SimpleFusionSearcher:
+    def __init__(self, index_dirs: List[str], method: FusionMethod):
+        self.method = method
+        self.searchers = [SimpleSearcher(index_dir) for index_dir in index_dirs]
+
+    def search(self, q: Union[str, JQuery], k: int = 10, query_generator: JQueryGenerator = None) -> TrecRun:
+        trec_runs = []
+        docid_to_search_result = dict()
+
+        for searcher in self.searchers:
+            hits = searcher.search(q, k, query_generator)
+            for hit in hits:
+                docid_to_search_result[hit.docid] = hit
+            run = TrecRun.generate_trec_run_from_search_results(hits)
+            trec_runs.append(run)
+
+        if self.method == FusionMethod.RRF:
+            fused_run = reciprocal_rank_fusion(trec_runs, rrf_k=60, depth=1000, k=1000)
+        else:
+            raise NotImplementedError()
+
+        return fused_run.map_trec_run_to_simple_search_result(docid_to_search_result)
+
+    def set_analyzer(self, analyzer):
+        for searcher in self.searchers:
+            searcher.set_analyzer(analyzer)
+
+    def set_rm3(self, fb_terms=10, fb_docs=10, original_query_weight=float(0.5), rm3_output_query=False):
```

Comment:
Remove all of these methods? Users can call `get_searchers` and configure individual searcher themselves?

Reply:
Oh, sounds good!